test/functional/terminal/tui_spec.lua

```diff
@@ -151,7 +151,7 @@ describe('tui', function()
   it('can handle arbitrarily long bursts of input', function()
     execute('set ruler')
     local t = {}
-    for i = 1, 3000 do
+    for i = 2000, 3000 do
```

Comment:
Thanks for figuring this out!  It definitely seems to be working now.  QB passed on both #4743 and #4737.